### ISSUE TITLE
RPC calls from plugins causing plugin killing

### ISSUE THREAD

**Issue by litch** [ElementsProject/lightning#5302]
### Issue and Steps to Reproduce

I compiled and tested the code from https://github.com/ElementsProject/lightning/blob/master/cln-rpc/examples/getinfo.rs 

That one works fine

If I run the same exact code from a plugin (see my fork of the example code), I get `UNUSUAL plugin-cln-plugin-startup: Killing plugin: JSON-RPC message does not contain "jsonrpc" field`.  

https://github.com/litch/lightning/blob/plugin-diagnosing/plugins/examples/cln-plugin-startup.rs#L34

Oddly!  Sometimes I actually get the `getinfo` back out correctly:

```
% lightningd/lightningd --network=regtest --log-level=DEBUG:plugin --plugin=/Users/litch/code/lightning/target/debug/examples/cln-plugin-startup
...
2022-06-02T00:09:04.556Z INFO    plugin-cln-plugin-startup: Read response Object({\"id\": Number(1), \"jsonrpc\": String(\"2.0\"), \"result\": Object({\"address\": Array([Object({\"address\": String(\"2605:a601:aa3d:3a00:88c3:98bf:5a56:3cb5\"), \"port\": Number(19846), \"type\": String(\"ipv6\")})]), \"alias\": String(\"STRANGEANALYST\"), \"binding\": Array([Object({\"address\": String(\"::\"), \"port\": Number(19846), \"type\": String(\"ipv6\")})]), \"blockheight\": Number(0), \"color\": String(\"03b289\"), \"fees_collected_msat\": String(\"0msat\"), \"id\": String(\"03b289234aa67ac25bf68f8907b5e77f80a6c03f893504ae189a11f6bcd6c89d6a\"), \"lightning-dir\": String(\"/Users/litch/.lightning/regtest\"), \"msatoshi_fees_collected\": Number(0), \"network\": String(\"regtest\"), \"num_active_channels\": Number(1), \"num_inactive_channels\": Number(0), \"num_peers\": Number(1), \"num_pending_channels\": Number(0), \"our_features\": Object({\"channel\": String(\"\"), \"init\": String(\"080269a2\"), \"invoice\": String(\"02000000024100\"), \"node\": String(\"800000080269a2\")}), \"version\": String(\"v0.11.0.1-13-gf90ee0e\"), \"warning_lightningd_sync\": String(\"Still loading latest blocks from bitcoind.\")})})
2022-06-02T00:09:04.557Z UNUSUAL plugin-cln-plugin-startup: Killing plugin: JSON-RPC message does not contain "jsonrpc" field
```

### `getinfo` output

```
% cli/lightning-cli --network=regtest getinfo
{
   "id": "03b289234aa67ac25bf68f8907b5e77f80a6c03f893504ae189a11f6bcd6c89d6a",
   "alias": "STRANGEANALYST",
   "color": "03b289",
   "num_peers": 1,
   "num_pending_channels": 0,
   "num_active_channels": 1,
   "num_inactive_channels": 0,
   "address": [
      {
         "type": "ipv6",
         "address": "2605:a601:aa3d:3a00:88c3:98bf:5a56:3cb5",
         "port": 19846
      }
   ],
   "binding": [
      {
         "type": "ipv6",
         "address": "::",
         "port": 19846
      }
   ],
   "version": "v0.11.0.1-13-gf90ee0e",
   "blockheight": 21104,
   "network": "regtest",
   "msatoshi_fees_collected": 0,
   "fees_collected_msat": "0msat",
   "lightning-dir": "/Users/litch/.lightning/regtest",
   "our_features": {
      "init": "080269a2",
      "node": "800000080269a2",
      "channel": "",
      "invoice": "02000000024100"
   }
}
```

**Comment by cdecker** [comment]
I'll need to look into this, do you run with `--allow-deprecated-apis=true`? That might temporarily allow getting the plugin working again until my fix lands.

**Comment by litch** [comment]
I can manually make the change I reference to the cln_rpc files and then build/use that artifact in my application so it’s not a blocker at the moment.  But it will just get clobbered next time I run ‘make clean’.

**Comment by litch** [comment]
I found the issue with this actually - @justinmoon mentioned at the conference today that it may be a logging issue.  And yep!  

Because I just copy/pasta'd in the `println!` statement, and the plugins use stdin, it was interpreting the `dbg!` and `println!` statements as rpc inputs and killing my plugin.  When I removed the comments, it all just works fine.

Also, this issue is unrelated to the cln_rpc not including the option_will_fund (which I think is what you were referring to, @cdecker in the above comment"?  I saw your PR related to msggen but have not had a chance to look at it.
